photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,342 @@
+                        self._fitshape = fitshape
+                    else:
+                        raise ValueError('fitshape must be odd integer-valued, '
+                                         'received fitshape = {}'\
+                                         .format(fitshape))
+                else:
+                    raise ValueError('fitshape must have positive elements, '
+                                     'received fitshape = {}'\
+                                     .format(fitshape))
+            else:
+                raise ValueError('fitshape must have two dimensions, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+                `~astropy.io.fits.HDUList`
+          Image to perform photometry
+        
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, ImageHDU, HDUList
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        return self.do_photometry(image)
+
+    def do_photometry(self, image):
+        # prepare output table
+        outtab = Table([[], [], [], [], [], []],
+                       names=('id', 'group_id', 'x_fit', 'y_fit', 'flux_fit',
+                              'iter_detected'),
+                       dtype=('i4', 'i4', 'f8', 'f8', 'f8', 'i4'))
+
+        # make a copy of the input image
+        residual_image = image.copy()
+
+        # perform background subtraction
+        residual_image = residual_image - self.bkg(image)
+
+        # find potential sources on the given image
+        sources = self.find(residual_image)
+
+        n = 1
+        # iterate until no more sources are found or the number of iterations
+        # has been reached
+        while(n <= self.niters and len(sources) > 0):
+            # prepare input table
+            intab = Table(names=['id', 'x_0', 'y_0', 'flux_0'],
+                          data=[sources['id'], sources['xcentroid'],
+                          sources['ycentroid'], sources['flux']])
+
+            # find groups of overlapping sources
+            star_groups = self.group(intab)
+
+            # fit the sources within in each group in a simultaneous manner
+            # and get the residual image
+            tab, residual_image = self.nstar(residual_image, star_groups)
+
+            # mark in which iteration those sources were fitted
+            tab['iter_detected'] = n*np.ones(tab['x_fit'].shape, dtype=np.int)
+
+            # populate output table
+            outtab = vstack([outtab, tab])
+
+            # find remaining sources in the residual image
+            sources = self.find(residual_image)
```

Comment:
This line should catch the warning: ``WARNING: No sources were found. [photutils.detection.findstars]``, as it's rather confusing to get the warning but obviously having sources detected in previous iterations.